### PR TITLE
Add assignee to task card

### PR DESCRIPTION
**Pull request by npendery**
# What's in this PR?

Add the ability to assign a user to a task

## References
Fixes #

Progress on: #

### CODE REVIEW

File: app/components/task-card-assignee-list.js [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+  run: { bind, once },
+  set
+} = Ember;
+
+export default Component.extend({
+  classNames: ['select-dropdown', 'task-assignee__users-list'],
+  tagName: 'div',
+  cursorAt: 0,
+  cursorWas: 0,
+  // hidden: false,
```
Comment (begedin) [inline app/components/task-card-assignee-list.js]: Guess you missed this.

Comment (begedin) [inline app/components/task-card-assignee-list.js]: Just a reminder to remove/uncomment this before merging

File: app/components/task-card-assignee-list.js [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+          set(this, 'hidden', false);
+          break;
+        case 'ArrowUp':
+          cursorAt = get(this, 'cursorAt');
+          this._setPosition(--cursorAt);
+          set(this, 'hidden', false);
+          break;
+        case 'Comma':
+        case 'Enter':
+          this._selectAssigneeSkill();
```
Comment (begedin) [inline app/components/task-card-assignee-list.js]: copy-paste remnant from skill list component, I guess. Can probably remove the section.

**Comment by JoshSmith** [comment]
- [ ] Remove hovering from `user-select` component
- [ ] Change target away from task card and to the element id itself

**Comment by JoshSmith** [comment]
I'm actually going to vote at this point that we circle back to create tests and clean up any code that was here before.

File: app/components/task/user/users-list-item.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    let user = get(this, 'user');
+    this.sendAction('selectUser', user);
+  },
+
+  mouseEnter() {
+    let user = get(this, 'user');
+    this.sendAction('hover', user);
+  }
+
+  // selectedChanged: observer('selected', function() {
```
Comment (npendery) [inline app/components/task/user/users-list-item.js]: Are we nixing this commented area or keeping it for v2?

File: app/components/task/user/users-list.js [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+    let store = get(this, 'store');
+    set(this, 'lastQuery', query);
+    store.query('user', { query, limit }).then((users) => {
+      set(this, 'results', users);
+      set(this, 'cursorAt', 0);
+      this._updateSelected();
+    });
+  },
+
+  async _selectUser(user) {
```
Comment (npendery) [inline app/components/task/user/users-list.js]: wait, what's this async business

File: app/components/task/user/users-list-item.js (hunk 3 above)
Comment (JoshSmith) [inline app/components/task/user/users-list-item.js]: We'll keep it for now commented but I do want to figure it out in the next time around.

File: app/components/task/user/users-list.js (hunk 4 above)
Comment (JoshSmith) [inline app/components/task/user/users-list.js]: We're doing a lot more work with `async`, which is coming in `es`.

**Comment by JoshSmith** [comment]
Follow-up issue #1064

**Pull request merged**